broker/broker.go

```diff
@@ -166,6 +179,42 @@ func (b *S3Broker) Deprovision(
 	return brokerapi.DeprovisionServiceSpec{IsAsync: false}, nil
 }
 
+func (b *S3Broker) getBucketNames(instanceNames []string, instanceGUID, planGUID string) ([]string, error) {
+	bucketNames := make([]string, len(instanceNames))
+
+	if len(instanceNames) == 0 {
+		return bucketNames, nil
+	}
+
+	instance, err := b.cfClient.ServiceInstanceByGuid(instanceGUID)
+	if err != nil {
+		return bucketNames, err
+	}
+
+	query := url.Values{}
+	query.Set("space_guid", instance.SpaceGuid)
+	query.Set("plan_guid", planGUID)
```

Comment:
Will this prevent you from making a credential that can access a basic and basic-public bucket?